src/main/java/seedu/address/logic/commands/AssignCommand.java

```diff
@@ -61,16 +59,14 @@ public CommandResult execute(Model model, OfficeConnectModel officeConnectModel)
 
         Person person = personList.get(personIndex.getZeroBased());
         Task task = taskList.get(taskIndex.getZeroBased());
-        Id pId = person.getId();
-        Id tId = task.getId();
-        AssignTask toAdd = new AssignTask(pId, tId);
 
-        RepositoryModelManager<AssignTask> personTaskModelManager = officeConnectModel.getAssignTaskModelManager();
-        if (personTaskModelManager.hasItem(toAdd)) {
+        AssignTask toAdd = new AssignTask(person, task);
+
+        if (officeConnectModel.hasAssignTaskModelManagerItem(toAdd)) {
             throw new CommandException(MESSAGE_DUPLICATE_ASSIGNMENT);
         }
 
-        personTaskModelManager.addItem(toAdd);
+        officeConnectModel.addAssignTaskModelManagerItem(toAdd);
```

Comment:
Good improvement of the code quality, command class now look much cleaner, and officeConnectModel start to function like what is suppose to.